src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -107,10 +102,11 @@ public Http2Connection(HttpConnectionPool pool, Stream stream)
 
             _httpStreams = new Dictionary<int, Http2Stream>();
 
-            _writerLock = new AsyncMutex();
             _connectionWindow = new CreditManager(this, nameof(_connectionWindow), DefaultInitialWindowSize);
             _concurrentStreams = new CreditManager(this, nameof(_concurrentStreams), int.MaxValue);
 
+            _writeChannel = Channel.CreateUnbounded<WriteQueueEntry>(new UnboundedChannelOptions() { SingleReader = true });
```

Comment:
Let's move the options into a singleton:
```C#
private static readonly UnboundedChannelOptions s_channelOptions = new UnboundedChannelOptions() { SingleReader = true };
...
_writeChannel = Channel.CreateUnbounded<WriteQueueEntry>(s_channelOptions);
```

Reply:
why it is a class instead of a struct? it only hold few bools and an enum

Reply:
Because there's a hierarchy of them and you're not expecting to create them very frequently.

Reply:
Done